core/src/test/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidatorTest.java

```diff
@@ -0,0 +1,217 @@
+package com.sequenceiq.cloudbreak.controller.validation.template;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.junit.Assert;
```

Comment:
The usage of JUnit 5 would be preferable; it is high time to get rid of JUnit 4. For assertions, please consider using AssertJ `assertThat()` etc. as its fluent API is more expressive. (Neither of these are mandatory or enforced anywhere, but it would be nice to use them in new tests at least.)